100_EDA_Cleaning/104-EDA.Rmd

```diff
@@ -0,0 +1,493 @@
+    races_df <- races_df[, !(names(races_df) %in% c("round", "url", "name"))]
+        
+    eda_results_level <- results_df %>% inner_join(races_df, by = c("raceId"))
+    
+    #join results with driver data
+    eda_results_level <- eda_results_level %>% inner_join(list_of_df[["drivers"]], by = c("driverId"))
+    eda_results_level$dob_driver <- eda_results_level$dob
+    eda_results_level <- eda_results_level[, !(names(eda_results_level) %in% c("number", "code", "forename", "surname", "url", "dob"))]
+    
+    #Translate date to date
+    eda_results_level$date <- as.Date(eda_results_level$date, "%Y-%m-%d")
+    eda_results_level$dob_driver <- as.Date(eda_results_level$dob_driver, "%Y-%m-%d")
+    
+    #NEW VAR: Create age at race
+    eda_results_level$age_at_race <- as.numeric(floor((eda_results_level$date - eda_race_level$dob) / 365.25))
+    
+    
+    #Join with drivers_standings on driverId + raceId 
+    driver_standings <-list_of_df[["driver_standings"]]
+    eda_results_level <- eda_results_level %>% left_join(driver_standings, by = c("raceId", "driverId"))
+    #TODO: any cols to drop?
+
+    #Join pit_stops_driver race on driverId + raceID to bring in Number of Stops
+    eda_results_level <- eda_results_level %>% left_join(pit_stops_driver_race, by = c("raceId", "driverId"))
+    
+    #Join status and StatusId
+    eda_results_level <- eda_results_level %>% left_join(list_of_df[["status"]], by = c("statusId"))
+    
+    #Join with constructors on team name
+    constructors <- list_of_df[["constructors"]]
+    constructors <-  constructors[, !(names(constructors) %in% c("url", "nationality", "constructorRef"))]
+    #Temp fixes for some teams
+    constructors$name[constructors$name == "Alpine F1 Team"] = "Alpine"
+    constructors$name[constructors$name == "Haas F1 Team"] = "Haas"
+    constructors$name[constructors$name == "AlphaTauri"] = "Alpha Tauri"
+    eda_results_level <- eda_results_level %>% left_join(constructors, by = c("constructorId"))
+    
+    constructor_standings <-list_of_df[["constructor_standings"]]
+    colnames(constructor_standings)[colnames(constructor_standings) == 'points'] <- 'constructor_points'
+    colnames(constructor_standings)[colnames(constructor_standings) == 'position'] <- 'constructor_position'
+    colnames(constructor_standings)[colnames(constructor_standings) == 'positionText'] <- 'constructor_positionText'
+    colnames(constructor_standings)[colnames(constructor_standings) == 'wins'] <- 'constructor_wins'
+    eda_reulst_level <- eda_results_level %>% left_join(constructor_standings, by = c("raceId", "constructorId"))
+    
+    eda_results_level <- eda_results_level %>% inner_join(eda_driver_level, by = c("year" = "Season", "Race_name" = "Race"))
+    
+   }
+  else{
+    eda_results_level
+  }
+
+}
+
+
+
+eda_results_level <- create_results_level_ds()
+summary(eda_results_level)
+
+
+```
+
+```{r}
+write.csv(eda_race_level, "../data/eda_results_level.csv", row.names=FALSE)
```

Comment:
Questions for discussion tomorrow:
1. Do we need to do Left Joins or Right Joins on the earlier dfs? I.e. have all drivers within a race with a 1/0 indicator of driver of the day?
2. are there any shared functions between 103 and 104 that we can move to an r script? Thinking both plots as well as readin/df joins.